Iss378

bechenard opened the pull request:
Ready to merge

JeremyMcCormick commented:
I don't think there should be a separate HcalStripHit class.  The existing HcalHit class should likely be modified with your changes.  It should work fine as long as the class version is incremented.  Was there some good reason not to do this?

bechenard commented:
HI Jeremy,

Sorry for the delayed reply, I don't check that mailbox often. Since Nhan
wanted to keep the old scheme to cross-validate the results, we decided to
go with separate products and modules for the time being. The trick is that
it is annoying to recover HcalHits from HcalStripHits as we randomly sample
the background in the strips or full plane.

We will likely replace the HcalHit with the HcalStripHit in the future
(when Nhan is happy).

Is that ok for you?
Cheers,
Bertrand


On Mon, Aug 28, 2017 at 1:17 PM, Jeremy McCormick <notifications@github.com>
wrote:

> I don't think there should be a separate HcalStripHit class. The existing
> HcalHit class should likely be modified with your changes. It should work
> fine as long as the class version is incremented. Was there some good
> reason not to do this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LDMXAnalysis/ldmx-sw/pull/379#issuecomment-325467961>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFOvpHyoezMcAj7uO8KrjzDH7CPl-KTDks5scyBAgaJpZM4PCOwV>
> .
>

nhanvtran commented:
Confirming...

nhanvtran commented on EventProc/src/HcalStripDigiProducer.cxx:
```diff
@@ -0,0 +1,141 @@
+    const float HcalStripDigiProducer::LAYER_ZWIDTH = 60.;
+    const int HcalStripDigiProducer::NUM_HCAL_LAYERS = 15;
+    const float HcalStripDigiProducer::MEV_PER_MIP = 1.40;
+    const float HcalStripDigiProducer::PE_PER_MIP = 13.5 * 6. / 4.;
```
So looking back at old talks, this should be 13.5 not 13.5 * 1.5...can we fix this and also in HcalDigiProducer.cxx?

nhanvtran commented on SimApplication/src/HcalSD.cxx:
```diff
@@ -43,23 +44,43 @@ namespace ldmx {
+        //         left/right side hcal: segmented along x direction every 10 cm
+        //         top/bottom side hcal: segmented along y direction every 10 cm
+        int stripID = -1;        
+        if (section==HcalSection::BACK) stripID = int( (localPosition.y()+scint->GetYHalfLength())/100.0);
```
Isn't this 1cm strips, not 10cm strips?

nhanvtran commented:
General comment: this will break the old layer-wise HCAL veto. 
The general scheme is sum over all simhits and aggregate them in the same detIDraw.  Then for a given detIDraw, make a hit.  Once we add the strips into the detIDraw, then one cannot do the same thing for layers.  

Should we update the layer-wise HCAL veto at the same time?

awhitbeck commented:
@nhanvtran I think we should make the changes to the old digi producer in this PR.  I think the simplest fix is to change `detIDraw` to `layer` for the map keys.   But I think this just effects the digi producers, not the veto processor.

@bechenard, I have a few minor comments for the `HcalStripDigiProducer`, some also apply equally to the `HcalDigiProducer`, which I can take care of, if you want:

 - This is wrong and probably should be removed:
https://github.com/LDMXAnalysis/ldmx-sw/blob/iss378/EventProc/src/HcalStripDigiProducer.cxx#L19-L21

 - Change layer to strip in comment: 
https://github.com/LDMXAnalysis/ldmx-sw/blob/iss378/EventProc/src/HcalStripDigiProducer.cxx#L83

 - Change var name to something more vague, or change layer to strip:
https://github.com/LDMXAnalysis/ldmx-sw/blob/iss378/EventProc/src/HcalStripDigiProducer.cxx#L83

I am still running some test, so there might be more to come, but after a quick look, it seems that everything for `HCalStripDigiProducer` works as expected.

nhanvtran commented:
@awhitbeck yes I agree, we should put the fix in this PR
@awhitbeck for your 3rd point, I think the link is wrong

awhitbeck commented:
I ran a test by generating muons which were randomly distributed in the x-y plane of the hcal.  I am seeing half the number of hits I would expect with strip id=30.  A plot is included which shows the hit multiplicity for all cells in the back section. 

![image](https://user-images.githubusercontent.com/4173436/30484773-805dde12-99f9-11e7-894a-fb617ab2ffba.png)

bechenard commented:
Weird I'll look into it. What are the bounds you use? 

Sent from my iPhone

> On Sep 15, 2017, at 7:40 AM, Andrew Whitbeck <notifications@github.com> wrote:
> 
> I ran a test by generating muons which were randomly distributed in the x-y plane of the hcal. I am seeing half the number of hits I would expect with strip id=30. A plot is included which shows the hit multiplicity for all cells in the back section.
> 
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

awhitbeck commented:
If I understand the code, its -1500,1500 mm in both x and y directions.

bechenard commented:
Hi Nhan,

Now, this is actually the correct definition. The strip 0 goes from -1550
to -1450, and if I have a point at -1500, then
localPosition.y()+scint->GetYHalfLength() = 50,
so dividing by 100 is actually the correct thing to do (dividing by 10
gives 1 cm strips).

On Wed, Sep 13, 2017 at 11:27 AM, nhanvtran <notifications@github.com>
wrote:

> *@nhanvtran* commented on this pull request.
> ------------------------------
>
> In SimApplication/src/HcalSD.cxx
> <https://github.com/LDMXAnalysis/ldmx-sw/pull/379#discussion_r138701689>:
>
> >          int section = copyNum / 1000;
> +        int layer = copyNum % 1000;
> +
> +        //stripID: back Hcal, segmented along y direction for now every 10 cm -- alternate x-y in the future?
> +        //         left/right side hcal: segmented along x direction every 10 cm
> +        //         top/bottom side hcal: segmented along y direction every 10 cm
> +        int stripID = -1;
> +        if (section==HcalSection::BACK) stripID = int( (localPosition.y()+scint->GetYHalfLength())/100.0);
>
> Isn't this 1cm strips, not 10cm strips?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LDMXAnalysis/ldmx-sw/pull/379#pullrequestreview-62532934>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFOvpEPAyZdlRC8qjlscLHi4DmPqKqL7ks5siB6bgaJpZM4PCOwV>
> .
>

bechenard commented:
Hi Andrew,

For some (weird) reason, this has been coded as -1550 -> 1550. This might
explain the deficit, but I'll run more checks.

On Fri, Sep 15, 2017 at 9:30 AM, Andrew Whitbeck <notifications@github.com>
wrote:

> If I understand the code, its -1500,1500 mm in both x and y directions.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LDMXAnalysis/ldmx-sw/pull/379#issuecomment-329832512>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFOvpDwb57VVuasQ-FVqYxsX9py29eOIks5siqYQgaJpZM4PCOwV>
> .
>

nhanvtran commented on EventProc/src/HcalDigiProducer.cxx:
```diff
@@ -12,108 +12,130 @@
+        } 
+        
+        
+        event.add("HcalDigis", hits_);
```
bug here...this should be `hcalDigis`, not `HcalDigis`

nhanvtran commented:
Besides the one bug above, it looks great.  I would add some example configuration for usres in
`Configuration/python/sim_recon_prod.py.tmpl`

The default values should be:
```
hcalDigis.parameters["meanNoise"] = 1.5
hcalDigis.parameters["mev_per_mip"] = 1.4
hcalDigis.parameters["pe_per_mip"] = 13.5
hcalDigis.parameters["doStrip"] = 0
```

bechenard commented:
Hi Nhan,

Should doStrip=0 be the default? Maybe for now.
I've fixed the bug and default values, and added a script to show how to use it (Configuration/python/sample_recon_Hcal.py) for free!

nhanvtran commented:
Sure, @bechenard thanks, everything looks good on my side.
@awhitbeck did you want to run another muon spray validation?  or should we merge this guy?
@omar-moreno @JeremyMcCormick is that ok?

bechenard commented:
Let Andrew spray first😀andrew could you post your script to do so

Thanks
Bertrand

Sent from my iPhone

> On Sep 20, 2017, at 7:22 AM, nhanvtran <notifications@github.com> wrote:
> 
> Sure, @bechenard thanks, everything looks good on my side.
> @awhitbeck did you want to run another muon spray validation? or should we merge this guy?
> @omar-moreno @JeremyMcCormick is that ok?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

awhitbeck commented:
Yep, I'll rerun the test and post a recipe.

omar-moreno commented:
Go for it.

On Wed, Sep 20, 2017 at 6:22 AM, nhanvtran <notifications@github.com> wrote:

> Sure, @bechenard <https://github.com/bechenard> thanks, everything looks
> good on my side.
> @awhitbeck <https://github.com/awhitbeck> did you want to run another
> muon spray validation? or should we merge this guy?
> @omar-moreno <https://github.com/omar-moreno> @JeremyMcCormick
> <https://github.com/jeremymccormick> is that ok?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LDMXAnalysis/ldmx-sw/pull/379#issuecomment-330849491>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADiWXFzc6NAHDPt4ZTQ8hHqzgPcSr2FEks5skRGMgaJpZM4PCOwV>
> .
>

awhitbeck commented:
Looks ready to go to me.
